packages/circuits/src/initial.circom

```diff
@@ -0,0 +1,61 @@
+pragma circom 2.0.0;
+
+include "./merkle.circom";
+
+template Initial(levels, cardCount) {
+    signal input deckRoot;
+    signal input newDeckRoot;
+    signal input deckLeaves[2**levels];
+    signal input newDeckLeaves[2**levels];
+    signal input newHandRoot;
+    signal input newHandLeaves[2**levels];
+    signal input deckPredicate[2**levels];
+    signal input drawnCardIndices[2**levels];
+
+    var maxDeckSize = 2**levels;
+
+    component checkInitialDeck = CheckMerkleRoot(levels);
+    checkInitialDeck.root <== deckRoot;
+    checkInitialDeck.leaves <== deckLeaves;
```

Comment:
We need to check the new deck root in the same way, no?

Also now that I think on it... This must already be a shuffled deck, right? Otherwise everyone knows what will be drawn.

This means we need another proof, that is a pure proof of shuffle wrt a known Merkle proof?
This is easy to implement (essentially, it's just a subset of this proof!).

This could potentially be done ahead of the game though, and could even be reused for multiple game.

I'm actually starting to think I'd love closed deck lists, but then a nasty side-effect is that the proof for every deck needs to be regenerated when cards are being traded (or moved out of the inventory).

Reply:
Yeap have now added a shuffle circuit. The workflow I am envisioning is:
1. opened deck list added
2. player shuffle the deck to hide the order of the cards, and committing a merkle root for the new order
3. random drawn indices generated by the contract, which is public, and the player draws the card based on these indices
4. player commits new deck root and new hand root

Just wanted to make sure I am thinking correctly?

Reply:
Oh yeah good catch hahaha, fixed the `newDeckRoot` issue.